src/common/rsvc.c

```diff
@@ -104,8 +104,22 @@ rsvc_client_process_error(struct rsvc_client *client, int rc,
 {
 	int leader_index = client->sc_leader_index;
 
-	if (client->sc_leader_known && client->sc_leader_aliveness > 0 &&
-	    ep->ep_rank == client->sc_ranks->rl_ranks[leader_index]) {
+	if (rc == -DER_NONEXIST) {
```

Comment:
if the server is one of the server replicas, but not the leader, it will return ENOTLEADER.
if the server is not one of the server replicas at all, it will return NONEXIST?

IMHO, both should return ENOTLEADER,  and use hint to adjust the rank list on client, probably would be clearer.
NONEXIST is too common error code.

Reply:
@wangdi1, we've added a new DAOS error code DER_NOTREPLICA, replacing DER_NONEXIST here, in libgurt.